### ISSUE TITLE
Snapshot creation fails with version 1.2.1

### ISSUE THREAD

**Issue by Mira-Roland** [tglunde/dbt-exasol#42]
Hello,
when upgrading from dbt-exasol 1.0.3 to version 1.2.1 an existing snapshot model failed to update.

I simplified the code for the snapshot creation as much as possible, but couldn't figure out the reason for the failure.
Just creating a simple new snapshot model I can not reproduce it, but here are the steps to create the error based on our existing snapshot model:

1. Create the snapshot table as an empty table:
```
CREATE TABLE TESTSCHEMA.MAWI_SNAPSHOT_ARTIKEL (
    ARTIKEL_HASHID HASHTYPE(16 BYTE),
    IK DECIMAL(10,0),
    STANDORT DECIMAL(10,0),
    EV_NUMMER DECIMAL(10,0),
    ARTIKEL_NUMMER VARCHAR(200) UTF8,
    ARTIKEL_BEZEICHNUNG VARCHAR(2000) UTF8,
    BEZUG_BEZEICHNUNG VARCHAR(200) UTF8,
    KA_HASHID HASHTYPE(16 BYTE),
    KA_NUMMER VARCHAR(200) UTF8,
    MENGENEINHEIT VARCHAR(100) UTF8,
    PZN VARCHAR(100) UTF8,
    WARENGRUPPE_BEZEICHNUNG VARCHAR(200) UTF8,
    LIEFERANT_BEZEICHNUNG VARCHAR(200) UTF8,
    ECLASS_CODED_NAME VARCHAR(50) UTF8,
    CREATED_ON TIMESTAMP,
    FILENAME VARCHAR(2000) UTF8,
    PROCESS_ID VARCHAR(50) UTF8,
    DUMMY_VALUE_FLAG DECIMAL(1,0),
    DUMMY_SOURCE_TABLE VARCHAR(20) ASCII,
    DBT_SCD_ID CHAR(32) ASCII,
    DBT_UPDATED_AT TIMESTAMP,
    DBT_VALID_FROM TIMESTAMP,
    DBT_VALID_TO TIMESTAMP
);
```

2. Create the source table for the snapshot:
```
CREATE TABLE TESTSCHEMA.MAWI_ARTIKEL_SOURCE (
		ARTIKEL_HASHID HASHTYPE(16 BYTE),
		EV_NUMMER DECIMAL(10,0),
		IK DECIMAL(10,0),
		STANDORT DECIMAL(10,0),
		ARTIKEL_NUMMER VARCHAR(200) UTF8,
		ARTIKEL_BEZEICHNUNG VARCHAR(2000) UTF8,
		ARTIKEL_BEZEICHNUNG_KURZ VARCHAR(2000) UTF8,
		BEZUG_NUMMER VARCHAR(200) UTF8,
		BEZUG_BEZEICHNUNG VARCHAR(2000) UTF8,
		BEZUG_BEZEICHNUNG_KURZ VARCHAR(2000) UTF8,
		KA_HASHID HASHTYPE(16 BYTE),
		KA_NUMMER VARCHAR(200) UTF8,
		KA_BEZEICHNUNG VARCHAR(2000) UTF8,
		KA_BEZEICHNUNG_KURZ VARCHAR(2000) UTF8,
		MENGENEINHEIT VARCHAR(50) UTF8,
		DARREICHUNGSFORM VARCHAR(50) UTF8,
		ECLASSCODEDNAME VARCHAR(50) UTF8,
		PZN VARCHAR(50) UTF8,
		GTIN_EAN_NUMMER VARCHAR(200) UTF8,
		WARENGRUPPE_NUMMER VARCHAR(200) UTF8,
		WARENGRUPPE_BEZEICHNUNG VARCHAR(2000) UTF8,
		WARENGRUPPE_BEZEICHNUNG_KURZ VARCHAR(2000) UTF8,
		CREATED_ON TIMESTAMP,
		FILENAME VARCHAR(200) UTF8,
		PROCESS_ID VARCHAR(50) UTF8
);
```

3. Create the snapshot model:
```
{% snapshot MAWI_Snapshot_Artikel %}

{{
    config(
      strategy='check',
      unique_key='ARTIKEL_HASHID',
      check_cols=['WARENGRUPPE_BEZEICHNUNG', 'MENGENEINHEIT', 'PZN', 'ECLASS_CODED_NAME',
                  'KA_HASHID', 'LIEFERANT_BEZEICHNUNG',],
      tags=["MAWI_Snapshot_Artikel"], 
    )
}}

SELECT DISTINCT
    art.ARTIKEL_HASHID             AS ARTIKEL_HASHID,
    art.IK                         AS IK,
    art.STANDORT                   AS STANDORT,
    art.EV_NUMMER                  AS EV_NUMMER,
    art.ARTIKEL_NUMMER             AS ARTIKEL_NUMMER,
    art.ARTIKEL_BEZEICHNUNG        AS ARTIKEL_BEZEICHNUNG,
    art.BEZUG_BEZEICHNUNG          AS BEZUG_BEZEICHNUNG,
    art.KA_HASHID                  AS KA_HASHID,
    art.KA_NUMMER                  AS KA_NUMMER,
    art.MENGENEINHEIT              AS MENGENEINHEIT,
    art.PZN                        AS PZN,
    art.WARENGRUPPE_BEZEICHNUNG    AS WARENGRUPPE_BEZEICHNUNG,
    'test'    AS LIEFERANT_BEZEICHNUNG,
    art.eClassCodedName            AS ECLASS_CODED_NAME,
    art.CREATED_ON                 AS CREATED_ON,
    art.FILENAME                   AS FILENAME,
    art.PROCESS_ID                 AS PROCESS_ID,
    1 AS DUMMY_VALUE_FLAG,
    'a' AS DUMMY_SOURCE_TABLE
FROM
    {{ ref('MAWI_ARTIKEL_SOURCE') }} art

{% endsnapshot %}
```

4. Run the snapshot:
```
dbt snapshot -m MAWI_Snapshot_Artikel
```

This is the output I get from dbt (the VARCHAR columns are all part of the output):
```
15:29:55  1 of 1 START snapshot USER_RGRUPE.MAWI_Snapshot_Artikel ........................ [RUN]
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
VARCHAR
15:29:59  1 of 1 ERROR snapshotting USER_RGRUPE.MAWI_Snapshot_Artikel .................... [ERROR in 4.06s]
15:29:59
15:29:59  Finished running 1 snapshot in 0 hours 0 minutes and 7.58 seconds (7.58s).
15:30:00
15:30:00  Completed with 1 error and 0 warnings:
15:30:00
15:30:00  Runtime Error in snapshot MAWI_Snapshot_Artikel (snapshots\cdw\mawi\MAWI_Snapshot_Artikel.sql)
15:30:00
15:30:00    (
15:30:00        message     =>  syntax error, unexpected UPDATE_, expecting END_OF_INPUT_ [line 4, column 5] (Session: 1750935480564252672)
15:30:00        dsn         =>  xxx.xxx.xxx.xxx..xxx:8563
15:30:00        user        =>  rgrupe
15:30:00        schema      =>
15:30:00        session_id  =>  1750935480564252672
15:30:00        code        =>  42000
15:30:00        query       =>  /* {"app": "dbt", "dbt_version": "1.2.1", "profile_name": "dbt-exasol", "target_name": "sbox", "node_id": "snapshot.miralytik.MAWI_Snapshot_Artikel"} */
15:30:00
15:30:00        alter table user_rgrupe.mawi_snapshot_artikel add column "WARENGRUPPE_BEZEICHNUNG__dbt_alter" VARCHAR(2000);
15:30:00        update user_rgrupe.mawi_snapshot_artikel set "WARENGRUPPE_BEZEICHNUNG__dbt_alter" = "WARENGRUPPE_BEZEICHNUNG";
15:30:00        alter table user_rgrupe.mawi_snapshot_artikel drop column "WARENGRUPPE_BEZEICHNUNG" cascade;
15:30:00        alter table user_rgrupe.mawi_snapshot_artikel rename column "WARENGRUPPE_BEZEICHNUNG__dbt_alter" to "WARENGRUPPE_BEZEICHNUNG"
15:30:00    )
15:30:00
15:30:00
15:30:00  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

I am really at a loss what is going on, so any help is appreciated.
Running the statements from the error message 1 by 1 on Exasol works just fine.
I am happy to provide any additional information necessary.

**Comment by tglunde** [comment]
@Mira-Roland , thank you for sharing this. We will be looking into this now. Would you be interested in participating during evaluation in case we get stuck?

**Comment by Mira-Roland** [comment]
Yes definitely, I will support where I can, just let me know if I can be of assistance.

**Comment by ilikutle** [comment]
Hello @Mira-Roland, as I understood, this error is showing up when the datatype of a column in Snapshot table is to be changed? I changed the macro, could you please checkout and install the dbt-exasol version from the pull request #45, test your case and let me know if this fixed the problem. Thank you.

**Comment by ilikutle** [comment]
@Mira-Roland, I removed this ugly output formatting, so if you can check it once again and let me know if it looks good so we can close the issue.
Thanks.

**Comment by Mira-Roland** [comment]
@ilikutle, yes, this fixed the output problem.
But even more so, thanks for pointing out the difference in the data type, I had totally missed that and it was not intended.
It also explains why I couldn't reproduce the problem with a simple snapshot model.
Either way I suppose it helped identify the problem, thanks again for your support!

**Comment by tglunde** [comment]
resolved with merge of #45
